codalab/lib/download_manager.py

```diff
@@ -57,6 +58,19 @@ def get_target_info(self, uuid, path, depth):
         bundle_state = self._bundle_model.get_bundle_state(uuid)
         # Raises NotFoundException if uuid is invalid
 
+        def check_dependencies(err):
```

Comment:
This is the core change

Reply:
Would be helpful to document the logic more thorougly with examples.